test/parallel/test-fs-mkdir.js

```diff
@@ -71,6 +77,89 @@ tmpdir.refresh();
   );
 });
 
+// mkdirpSync when both top-level, and sub-folders do not exist.
+{
+  const pathname = `${tmpdir.path}/${nextdir()}/${nextdir()}`;
+
+  fs.mkdirSync(pathname, { createParents: true });
+
+  const exists = common.fileExists(pathname);
+  assert.strictEqual(exists, true);
+  assert.strictEqual(fs.statSync(pathname).isDirectory(), true);
+}
+
+// mkdirpSync when folder already exists.
+{
+  const pathname = `${tmpdir.path}/${nextdir()}/${nextdir()}`;
+
+  fs.mkdirSync(pathname, { createParents: true });
+  // should not cause an error.
+  fs.mkdirSync(pathname, { createParents: true });
+
+  const exists = common.fileExists(pathname);
+  assert.strictEqual(exists, true);
+  assert.strictEqual(fs.statSync(pathname).isDirectory(), true);
+}
+
+// mkdirpSync ../
+{
+  const pathname = `${tmpdir.path}/${nextdir()}/../${nextdir()}/${nextdir()}`;
+  fs.mkdirSync(pathname, { createParents: true });
+  const exists = common.fileExists(pathname);
+  assert.strictEqual(exists, true);
+  assert.strictEqual(fs.statSync(pathname).isDirectory(), true);
+}
+
+// mkdirpSync when path is a file.
+{
+  const pathname = `${tmpdir.path}/${nextdir()}/${nextdir()}`;
+
+  fs.mkdirSync(path.dirname(pathname));
+  fs.writeFileSync(pathname, '', 'utf8');
+
+  try {
+    fs.mkdirSync(pathname, { createParents: true });
+    throw new Error('unreachable');
+  } catch (err) {
+    assert.notStrictEqual(err.message, 'unreachable');
+    assert.strictEqual(err.code, 'EEXIST');
+    assert.strictEqual(err.syscall, 'mkdir');
+  }
+}
+
+// mkdirp when folder does not yet exist.
+{
+  const pathname = `${tmpdir.path}/${nextdir()}/${nextdir()}`;
+
+  fs.mkdir(pathname, { createParents: true }, common.mustCall(function(err) {
+    assert.strictEqual(err, null);
+    assert.strictEqual(common.fileExists(pathname), true);
+    assert.strictEqual(fs.statSync(pathname).isDirectory(), true);
+  }));
+}
+
+// mkdirp when path is a file.
+{
+  const pathname = `${tmpdir.path}/${nextdir()}/${nextdir()}`;
+
+  fs.mkdirSync(path.dirname(pathname));
+  fs.writeFileSync(pathname, '', 'utf8');
+  fs.mkdir(pathname, { createParents: true }, (err) => {
+    assert.strictEqual(err.code, 'EEXIST');
+    assert.strictEqual(err.syscall, 'mkdir');
+    assert.strictEqual(fs.statSync(pathname).isDirectory(), false);
+  });
+}
+
+// mkdirpSync dirname loop.
+{
+  const pathname = `${tmpdir.path}/${nextdir()}`;
+  fs.mkdirSync(pathname);
+  process.chdir(pathname);
+  fs.rmdirSync(pathname);
+  fs.mkdirSync('X', { createParents: true }); // should not loop infinitely.
```

Comment:
I think this should throw `ENOENT`, just like the `createParents: false` version does?